BUILD

```diff
@@ -27,6 +37,181 @@ rust_binary(
+)
+
+assemble_targz(
+    name = "assemble-linux-arm64-targz",
+    additional_files = assemble_files,
+    empty_directories = empty_directories,
+    output_filename = "typedb-all-linux-arm64",
+    permissions = permissions,
+    targets = ["//:package-typedb"],
+    visibility = ["//tests/assembly:__subpackages__"],
+    target_compatible_with = constraint_linux_arm64,
+)
+
+deploy_artifact(
+    name = "deploy-mac-x86_64-zip",
+    artifact_group = "typedb-all-mac-x86_64",
+    artifact_name = "typedb-all-mac-x86_64-{version}.zip",
+    release = deployment["artifact"]["release"]["upload"],
+    snapshot = deployment["artifact"]["snapshot"]["upload"],
+    target = ":assemble-mac-x86_64-zip",
+)
+
+deploy_artifact(
+    name = "deploy-mac-arm64-zip",
+    artifact_group = "typedb-all-mac-arm64",
+    artifact_name = "typedb-all-mac-arm64-{version}.zip",
+    release = deployment["artifact"]["release"]["upload"],
+    snapshot = deployment["artifact"]["snapshot"]["upload"],
+    target = ":assemble-mac-arm64-zip",
+)
+
+deploy_artifact(
+    name = "deploy-linux-x86_64-targz",
+    artifact_group = "typedb-all-linux-x86_64",
+    artifact_name = "typedb-all-linux-x86_64-{version}.tar.gz",
+    release = deployment["artifact"]["release"]["upload"],
+    snapshot = deployment["artifact"]["snapshot"]["upload"],
+    target = ":assemble-linux-x86_64-targz",
+)
+
+deploy_artifact(
+    name = "deploy-linux-arm64-targz",
+    artifact_group = "typedb-all-linux-arm64",
+    artifact_name = "typedb-all-linux-arm64-{version}.tar.gz",
+    release = deployment["artifact"]["release"]["upload"],
+    snapshot = deployment["artifact"]["snapshot"]["upload"],
+    target = ":assemble-linux-arm64-targz",
+)
+
+# Convenience
+alias(
+    name = "assemble-typedb-server",
+    actual = select({
+        "@vaticle_bazel_distribution//platform:is_linux_arm64" : ":assemble-linux-arm64-targz",
+        "@vaticle_bazel_distribution//platform:is_linux_x86_64" : ":assemble-linux-x86_64-targz",
+        "@vaticle_bazel_distribution//platform:is_mac_arm64" : ":assemble-mac-arm64-zip",
+        "@vaticle_bazel_distribution//platform:is_mac_x86_64" : ":assemble-mac-x86_64-zip",
+#        "@vaticle_bazel_distribution//platform:is_windows_x86_64" : ":assemble-windows-x86_64-zip"
+    }),
+    visibility = ["//tests/assembly:__subpackages__"],
+)
+alias(
+    name = "deploy-typedb-server",
+    actual = select({
+        "@vaticle_bazel_distribution//platform:is_linux_arm64" : ":deploy-linux-arm64-targz",
+        "@vaticle_bazel_distribution//platform:is_linux_x86_64" : ":deploy-linux-x86_64-targz",
+        "@vaticle_bazel_distribution//platform:is_mac_arm64" : ":deploy-mac-arm64-zip",
+        "@vaticle_bazel_distribution//platform:is_mac_x86_64" : ":deploy-mac-x86_64-zip",
+#        "@vaticle_bazel_distribution//platform:is_windows_x86_64" : ":deploy-windows-x86_64-zip"
+    })
+)
+
+# docker
+docker_container_image(
+    name = "assemble-docker",
```

Comment:
The other thing we should look into is an ARM docker image - check you check if the bazel rules support arm docker images yet? last time i checked a year ago they didn't...